watch.js

```diff
@@ -453,32 +453,77 @@ function handleReply(self, msg) {
 	delete self.reply[msg.d.r];
 }
 
-function handleUpdate(self, path, newData) {
-	path = path.split('/');
+function _convertObjToLevels(inputObj) {
+
+	if (!inputObj || (typeof inputObj !== 'object')) {
+		return inputObj;
+	}
+
+	var resultObj = {};
+	convertHelper(inputObj, []);
+	return resultObj;
+
+	function convertHelper(currentRawObj, path) {
+
+		// If the current raw object is a scalar or just the empty object, navigate to
+		// where this raw object should live by walking down the path in resultObj, and
+		// put this raw object there.
+		if ((typeof currentRawObj !== 'object') || (!Object.keys(currentRawObj).length)) {
```

Comment:
Potential performance optimization: we'll end up enumerating all of the keys of the object twice -- here and at 492.

Since firebase doesn't support empty objects, can we avoid the `Object.keys(currentRawObj).length` altogether?